### ISSUE TITLE
Implement dpop for access token

### ISSUE THREAD

**Issue by tung2744** [authgear/authgear-server#4320]

**Comment by louischan-oursky** [comment]
We may want to DPoP bind the refresh_token only. See https://www.rfc-editor.org/rfc/rfc9449#section-5-9 My concern is that the access token may be used in a way that is assumed to be Bearer.

Possible configuration
DPoP:
- Disabled. DPoP is disabled
- Except access_token. DPoP is enabled for authorization code, refresh token, device secret, and device browser session token.
- All. DPoP is enabled for authorization code, access token, refresh token, device secret, and device browser session token.

**Comment by tung2744** [comment]
> My concern is that the access token may be used in a way that is assumed to be Bearer.

@louischan-oursky Yes. This is why I marked this issue as TBC. If we implement it, we need to support the DPoP Authentication Scheme. As access tokens are also read by resource servers, it could break existing integrations.

So what I plan to support is
- When DPoP header is present on /token, these tokens are DPoP bound:
    - refresh_token
    - device_secret
    - x_device_browser_session_token

And it solely depends on whether DPoP header is set on making the /token request. If it is present, the returned tokens are DPoP bound, else they are not.

**Comment by louischan-oursky** [comment]
> it solely depends on whether DPoP header is set on making the /token request.

The client metadata should also be implemented. https://www.rfc-editor.org/rfc/rfc9449#name-client-registration-metadat
